kratos/utilities/geometry_utilities.h

```diff
@@ -55,6 +55,114 @@ class KRATOS_API(KRATOS_CORE) GeometryUtils
      */
     static std::string GetGeometryName(const GeometryData::KratosGeometryType TypeOfGeometry);
 
+    /**
+     * @brief Returns the local coordinates of a given arbitrary point for a given linear tetrahedra
+     * @details Based on https://www.colorado.edu/engineering/CAS/courses.d/AFEM.d/AFEM.Ch09.d/AFEM.Ch09.pdf. Section 9.1.6
+     * @param rGeometry The geometry to be considered
+     * @param rResult The vector containing the local coordinates of the point
+     * @param rPoint The point in global coordinates
+     * @return The vector containing the local coordinates of the point
+     */
+    template<class TGeometryType>
+    static inline typename TGeometryType::CoordinatesArrayType& PointLocalCoordinatesTetrahedra3D4N(
+        const TGeometryType& rGeometry,
+        typename TGeometryType::CoordinatesArrayType& rResult,
+        const typename TGeometryType::CoordinatesArrayType& rPoint
+        )
+    {
+        // Compute RHS
```

Comment:
At least in debug, I'd check that `rGeometry` is of tetrahedra type.

Reply:
Okay